k9mail/src/main/java/com/fsck/k9/Preferences.java

```diff
@@ -164,4 +164,17 @@ public void setDefaultAccount(Account account) {
     public SharedPreferences getPreferences() {
         return mStorage;
     }
+
+    public static <T extends Enum<T>> T getEnumStringPref(SharedPreferences prefs, String key, T defaultEnum) {
+        String stringPref = prefs.getString(key, defaultEnum.name());
+        try {
+            return Enum.valueOf(defaultEnum.getDeclaringClass(), stringPref);
+        } catch (Exception ex) {
```

Comment:
On a project with more test automation, I would be more gung-ho about allowing exceptions to trickle up. But in this case, I was moving some existing code that already caught Exception.

Anyway, changed to IllegalArgumentException.